service/db/src/main/java/gov/va/vro/service/db/SaveToDbServiceImpl.java

```diff
@@ -45,8 +44,12 @@ public Claim insertClaim(Claim claim) {
 
   @Override
   public void insertAssessmentResult(
-      UUID claimId, AbdEvidenceWithSummary evidenceResponse, String diagnosticCode) {
-    ClaimEntity claimEntity = claimRepository.findById(claimId).orElse(null);
+      String claimSubmissionId,
+      String idType,
+      AbdEvidenceWithSummary evidenceResponse,
+      String diagnosticCode) {
+    ClaimEntity claimEntity =
+        claimRepository.findByClaimSubmissionIdAndIdType(claimSubmissionId, idType).orElse(null);
```

Comment:
I don't see why convert to null here. You can just check if the optional is present